python/tests/test_dataset.py

```diff
@@ -545,6 +547,13 @@ def test_dataset_proxy_set_variance():
     np.testing.assert_array_equal(d["a"].variances, variances)
 
 
+def test_rename_dims():
+    a = make_simple_dataset(Dim.Y, seed=0)
+    b = make_simple_dataset(Dim.Z, seed=0)
+    a.rename_dims({Dim.Y: Dim.Z})
```

Comment:
Maybe could add a test where you rename more than one dim at the same time?
Either here or in the C++ code.

Reply:
I guess that means I'll also need to fix the shortcoming of not being able to swap dimension. `rename_dims({Dim.Y: Dim.X, Dim.X: Dim.Y})` is currently not supported. :(

Reply:
This is too complicated now since not urgently need. I'll create a new task for it.